src/components/ButtonWithDropdownMenu/index.tsx

```diff
@@ -65,50 +66,68 @@ function ButtonWithDropdownMenu<IValueType>({
         }
     }, [windowWidth, windowHeight, isMenuVisible, anchorAlignment.vertical]);
 
+    const getIconRightButton = useCallback(
+        () => (
+            <Icon
+                medium={isButtonSizeLarge}
+                small={!isButtonSizeLarge}
+                src={Expensicons.DownArrow}
+                fill={success ? theme.buttonSuccessText : theme.icon}
+            />
+        ),
+        [isButtonSizeLarge, success, theme.buttonSuccessText, theme.icon],
+    );
+
     return (
         <View style={wrapperStyle}>
             {shouldAlwaysShowDropdownMenu || options.length > 1 ? (
                 <View style={[styles.flexRow, styles.justifyContentBetween, styles.alignItemsCenter, style]}>
                     <Button
                         success={success}
                         pressOnEnter={pressOnEnter}
-                        ref={buttonRef}
-                        onPress={(event) => onPress(event, selectedItem.value)}
+                        ref={(ref) => {
+                            caretButton.current = ref;
+                        }}
+                        onPress={(event) => (!isSplit ? setIsMenuVisible(!isMenuVisible) : onPress(event, selectedItem.value))}
                         text={customText ?? selectedItem.text}
                         isDisabled={isDisabled || !!selectedItem.disabled}
                         isLoading={isLoading}
                         shouldRemoveRightBorderRadius
                         style={[styles.flex1, styles.pr0]}
                         large={isButtonSizeLarge}
                         medium={!isButtonSizeLarge}
-                        innerStyles={[innerStyleDropButton, customText !== undefined && styles.cursorDefault, customText !== undefined && styles.pointerEventsNone]}
+                        innerStyles={[innerStyleDropButton, !isSplit && styles.dropDownButtonCartIconView]}
```

Comment:
```suggestion
                        innerStyles={[innerStyleDropButton, !isSplit && styles.dropDownButtonCaretIconView]}
```

Reply:
Why do we need this specific style? Can't we keep `customText !== undefined && styles.cursorDefault, customText !== undefined && styles.pointerEventsNone`?

Reply:
Previously, the button could only be clicked on the icon to display a dropdown menu and this style was to prevent users from being able to click on most parts of the button.
In this PR, we allow users to click on the large part of the button to display a dropdown.